test.py

```diff
@@ -1684,7 +1684,63 @@ def write_consolidated_boost_junit_xml(tmpdir: str, mode: str) -> None:
             test = ET.SubElement(suite, 'TestSuite', name=test_name)
         test.append(test_case)
     et = ET.ElementTree(xml)
-    et.write(f'{tmpdir}/{mode}/xml/boost.xunit.xml', encoding='unicode')
+    xunit_file = f'{tmpdir}/{mode}/xml/boost.xunit.xml'
+    et.write(xunit_file, encoding='unicode')
+    return xunit_file
+
+
+def boost_to_junit(boost_xml, junit_xml):
+    boost_root = ET.parse(boost_xml).getroot()
+    junit_root = ET.Element('testsuites')
+
+    test_suites = boost_root.find('TestSuite').findall("TestSuite")
+
+    def parse_tag_output(test_case_element: ET.Element, tag_output: str) -> str:
+        text_template = """
+            [{level}] - {level_message}
+            [FILE] - {file_name}
+            [LINE] - {line_number}
+            """
+        text = ''
+        tag_outputs = test_case_element.findall(tag_output)
+        if len(tag_outputs) > 0:
```

Comment:
Is this `if` necessary? If we get an empty list, the `for` below should be a no-op

Reply:
You are completely right.